### PR TITLE
Add utilities to calculate Order Parameter

### PR DESCRIPTION
**Pull request by jennyfothergill**
My goal is to calculate the order parameter shown in [doi:10.3390/polym10121305](https://doi.org/10.3390/polym10121305).
I will reference [Evan's code](https://bitbucket.org/cmelab/evan_analysis_scripts/src/master/) where possible.
The goal is to have a function which can calculate the clustering based on a distance cutoff and angle cutoff between the plane normal vectors of the chromophores.
Will use GRiTS to get the chromophores.

TODO

- [x] unit tests

### CODE REVIEW

File: cmeutils/geometry.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    assert points.shape[0] >= 3, "Need at least 3 points to calculate a plane."
+    ctr = points.mean(axis=0)
+    shiftpoints = points - ctr
+    U, S, Vt = svd(shiftpoints.T @ shiftpoints)
+    normal = U[:, -1]
+    return ctr, normal
+
+
+def angle_between_vectors(u, v):
```
Comment (chrisjonesBSU) [inline cmeutils/geometry.py:63]: This seems like it would be a useful function to finding bond angles, but in that case you might want to allow angles greater than 90 and instead set the limit to 180. Could we add a parameter for setting a max angle constraint, and the function contains different logic depending on that parameter?

File: cmeutils/gsd_utils.py [hunk 2]
```diff
@@ -133,3 +133,54 @@ def _validate_inputs(gsd_file, snap, gsd_frame):
...
+        i for i in snap.particles.types if i not in delete_types
+    ]
+    typeid_map = {
+        i:new_snap.particles.types.index(e)
+        for i, e in enumerate(snap.particles.types)
+        if e in new_snap.particles.types
+    }
+    new_snap.particles.position = snap.particles.position[selection]
+    new_snap.particles.image = snap.particles.image[selection]
+    new_snap.particles.typeid = np.vectorize(typeid_map.get)(
```
Comment (chrisjonesBSU) [inline cmeutils/gsd_utils.py:173]: Maybe this could be a separate PR, but I feel like we might as well write out all of the information of `snap.particles` available attributes? 

```
'angmom', 'body', 'charge', 'diameter', 'image', 'mass', 'moment_inertia', 'orientation', 'position', 'type_shapes', 'typeid', 'types', 'validate', 'velocity'
```

File: cmeutils/geometry.py (hunk 1 above)
Comment (jennyfothergill) [reply cmeutils/geometry.py:63]: yes, agreed that should be an option 👍

File: cmeutils/gsd_utils.py (hunk 2 above)
Comment (jennyfothergill) [reply cmeutils/gsd_utils.py:173]: Yes, agreed--let's let this be a separate PR though. I ended up not using this function anyway since I use grits c:
